jetbrains-core/src/software/aws/toolkits/jetbrains/ui/SliderPanel.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui;
+
+import com.intellij.openapi.ui.ValidationInfo;
+import org.jetbrains.annotations.TestOnly;
+
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.JTextField;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+
+import static software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.DEFAULT_MEMORY_SIZE;
+import static software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.DEFAULT_TIMEOUT;
+import static software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MAX_MEMORY;
+import static software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MAX_TIMEOUT;
+import static software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MEMORY_INCREMENT;
+import static software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MIN_MEMORY;
+import static software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MIN_TIMEOUT;
+import static software.aws.toolkits.resources.Localization.message;
+
+// A panel with a slider and text field, of which the slider and text field always synced up.
+public class SliderPanel {
+    private final int min, max, minTick, maxTick, minorTick, majorTick, defaultValue;
+    private final boolean snap;
+
+    private JPanel content;
+    public JSlider slider;
+    public JTextField textField;
+
+    public static SliderPanel lambdaTimeout() {
```

Comment:
This definitely doesn't live here - the `jetbrains.ui` stuff should be pretty much service agnostic

Reply:
Moved this to `Lambda` file.